Unify spec file organization

mvz opened the pull request:
## Summary

Re-organize feature files to match master

## Details

Files were moved and a few specs added.

## Motivation and Context

Part of #598.

## How Has This Been Tested?

N/A. Functionality should stay the same.

## Types of changes

- Refactoring (cleanup of codebase without changing any existing functionality)

luke-hill commented on spec/aruba/api/core_spec.rb:
```diff
@@ -0,0 +1,138 @@
+      it { expect { @aruba.expand_path('') }.to raise_error ArgumentError }
+    end
+
+    context 'when dir_path is given similar to File.expand_path ' do
```
Is this not covered by the other tests; it's a bit tricky for a newbie to see though, so ignore me if I'm wrong.

Also rogue space at the end

luke-hill commented on spec/aruba/api/core_spec.rb:
```diff
@@ -0,0 +1,138 @@
+        allow(config).to receive(:working_directory).and_return aruba.config.working_directory
+      end
+
+      before :each do
```
Not sure on styling, but to separate the blocks I'd just leave a line-space, not new blocks for them all. Allows you to save on 5-10 lines in this file

mvz replied:
Agreed! I'll also clean this up in master.

luke-hill commented on spec/aruba/api/core_spec.rb:
```diff
@@ -0,0 +1,138 @@
+      end
+
+      before :each do
+        allow(environment).to receive(:clear)
```
Even if this returns `nil` I'd put it in. Because otherwise this allows errors to spawn (Not sure on likelihood)

mvz replied:
I'm think I'm going to rewrite this spec because it contains way too much mocking. I copied it straight from master so I'll change it there also.

luke-hill commented on spec/aruba/api/core_spec.rb:
```diff
@@ -0,0 +1,138 @@
+      it 'does not touch other environment variables in the passed block' do
+        @aruba.create_directory @directory_name
+        @aruba.cd @directory_name do
+          expect(ENV['HOME']).not_to be_nil
```
this should probably have a 2nd assertion, as this doesn't really prove the assertion on line 117

luke-hill commented on spec/aruba/api/deprecated_spec.rb:
```diff
@@ -305,4 +311,106 @@ def actual_permissions
+      it { expect(ENV['LONG_LONG_ENV_VARIABLE']).to eq '2' }
+    end
+
+    context 'when existing variable by outer context' do
```
Does this use shared_contexts, otherwise this isn't testing anything.

luke-hill commented on spec/aruba/api/deprecated_spec.rb:
```diff
@@ -305,4 +311,106 @@ def actual_permissions
+          @aruba.restore_env
+        end
+
+        it { expect(ENV).not_to be_key 'LONG_LONG_ENV_VARIABLE' }
```
should this be `not_to have_key` (Same for below)

luke-hill commented on spec/aruba/api/deprecated_spec.rb:
```diff
@@ -305,4 +311,106 @@ def actual_permissions
+    end
+  end
+
+  describe "#restore_env" do
```
spacing in this describe block

luke-hill commented on spec/aruba/api/filesystem_spec.rb:
```diff
@@ -0,0 +1,766 @@
+
+    context 'when file does not exist' do
+      let(:name) { 'non_existing_file' }
+      it { expect { size }.to raise_error RSpec::Expectations::ExpectationNotMetError }
```
This doesn't make sense to me; shouldn't it be an NME?

luke-hill commented on spec/aruba/api/filesystem_spec.rb:
```diff
@@ -0,0 +1,766 @@
+      end
+    end
+
+    context "checking the file's content against the expectations in the block" do
```
probably should elude to regex here in this context

mvz replied:
These tests were moved from another file, but I don't think they add anything so I'm going to remove them.

luke-hill commented on spec/aruba/api/filesystem_spec.rb:
```diff
@@ -0,0 +1,766 @@
+            expect(full_content).to     match(/foo/)
+            expect(full_content).not_to match(/zoo/)
+          end
+        end . not_to raise_error
```
spacing here and below

luke-hill commented on spec/aruba/processes/spawn_process_spec.rb:
```diff
@@ -42,19 +42,29 @@
-      it { process.stop }
+      it { expect { process.stop }.not_to raise_error }
+
+      it 'makes the process stopped' do
```
`makes the process stop`

luke-hill commented on spec/aruba/processes/spawn_process_spec.rb:
```diff
@@ -42,19 +42,29 @@
     context "when process run succeeds" do
       it { expect { process.start }.not_to raise_error }
+
+      it 'makes the process started' do
```
`starts the process`

olleolleolle commented on spec/aruba/api/core_spec.rb:
```diff
@@ -0,0 +1,138 @@
+require 'aruba/api'
+require 'fileutils'
+
+describe Aruba::Api::Core do
```
Can we use `RSpec.describe`?

```suggestion
RSpec.describe Aruba::Api::Core do
```

mvz commented:
Thanks for your reviews, @luke-hill and @olleolleolle. I think I addressed all your issues.

(Spec style is a bit of an issue since a lot of specs use `it` blocks without descriptions which is definitely not my favourite.)

olleolleolle commented:
@mvz Looks like an ActiveSupport-ism is in there:

```
     NoMethodError:
       undefined method `slice' for #<Object:0x0000000000dbff38>```

mvz commented:
ActiveSupport :sob:

mvz commented:
> Looks like an ActiveSupport-ism is in there

Oh, I think it's actually part of standard Ruby now, that's why I didn't notice this locally.

olleolleolle commented:
@mvz https://blog.bigbinary.com/2018/02/06/ruby-2-5-added-hash-slice-method.html But - the `ENV` may not support it?

```
ENV.slice("2.5.1 :001 > ENV.slice("OLLE")
Traceback (most recent call last):
       16: from /Users/olle/.rvm/gems/ruby-2.5.1/bin/bundle:23:in `load'
       15: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/exe/bundle:22:in `<top (required)>'
       14: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
       13: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/exe/bundle:30:in `block in <top (required)>'
       12: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/cli.rb:18:in `start'
       11: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
       10: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/cli.rb:27:in `dispatch'
        9: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
        8: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
        7: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        6: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/cli.rb:463:in `exec'
        5: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/cli/exec.rb:28:in `run'
        4: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/cli/exec.rb:74:in `kernel_load'
        3: from /Users/olle/.rvm/gems/ruby-2.5.1/gems/bundler-1.17.3/lib/bundler/cli/exec.rb:74:in `load'
        2: from /Users/olle/.rvm/rubies/ruby-2.5.1/bin/irb:11:in `<top (required)>'
        1: from (irb):1
NoMethodError (undefined method `slice' for #<Object:0x00007ffda3828ea0>)
2.5.1 :002 > ENV.to_h.slice("OLLE")
{}```

mvz commented:
> But - the ENV may not support it?

I guess that was added in 2.5 then :man_shrugging:.

mvz commented:
This failure right there is also super weird: https://travis-ci.org/cucumber/aruba/jobs/525669344#L677

luke-hill commented on spec/aruba/api/core_spec.rb:
```diff
@@ -80,7 +80,8 @@
 
       it 'does not touch other environment variables in the passed block' do
-        keys = ENV.keys - ['PWD', 'OLDPWD']
+        # FIXME: Shouldn't have to exclude JRUBY_OPTS and JAVA_OPTS
```
Can we try copy something like what Selenium does for other committers benefit, it also pops up as a flag in the IntelliJ IDEs

Something like `#TODO: mvz - Shouldn't have to....` or `#FIXME: TODO - Mathis - Shouldn't have to...`

mvz replied:
I've instead create a fix for the problem in #610.

luke-hill commented on spec/aruba/api/deprecated_spec.rb:
```diff
@@ -305,4 +311,111 @@ def actual_permissions
+        ENV.delete 'LONG_LONG_ENV_VARIABLE'
+      end
+
+      context 'when set single' do
```
`'when set once'` or `'when set a single time'`

luke-hill commented on spec/aruba/api/deprecated_spec.rb:
```diff
@@ -305,4 +311,111 @@ def actual_permissions
+        ENV['LONG_LONG_ENV_VARIABLE'] = '0'
+      end
+
+      context 'when set single time' do
```
`'when set once'` or `'when set a single time'`

luke-hill commented on spec/aruba/api/filesystem_spec.rb:
```diff
@@ -0,0 +1,746 @@
+  context '#write_fixed_size_file' do
+    it "should write a fixed sized file" do
+      @aruba.write_fixed_size_file(@file_name, @file_size)
+      expect(File.exist?(@file_path)).to eq true
```
this style conflicts with line235. Both are valid, but should probably only use one style?

mvz replied:
That would be the end goal, but it'll be a long process :slightly_smiling_face:

mvz commented:
Ow, if you cancel an allowed-failure build Travis still reports a build failure :disappointed:.

Anyway, thanks @olleolleolle and @luke-hill for your work on reviewing this. I'm going to merge this now as I have addressed all your comments
